web-ui/src/main/resources/catalog/components/viewer/heatmap/HeatmapService.js

```diff
@@ -58,6 +58,7 @@
           return value * BUFFER_RATIO;
         });
         var extent = map.getView().calculateExtent(bufferedSize);
+        extent = ol.proj.transformExtent(extent, "EPSG:3857", "EPSG:4326");
```

Comment:
This will not work if the view is not in web mercator. Something like `map.getView().getProjection()` is needed.